### PR TITLE
release bugs, sporder message and 0 padding

### PR DESCRIPTION
**Pull request by adelebai**
release bugs, sporder message and 0 padding

### CODE REVIEW

File: util.hpp [hunk 1]
```diff
@@ -314,4 +314,14 @@ vector<string> getFileHeaders(const string& header, const string& filename) {
     return headers;
 }
 
+// converts number to a string form, but with padding
+// does nothing if stringLength is longer than digits in number
```
Comment (andrist) [inline util.hpp]: // converts number to a string padded with leading zeros
// does nothing if stringLength is less than the digits in number.]

File: test/utils_test.cpp [hunk 2]
```diff
@@ -140,4 +140,13 @@ TEST(UtilTestsGroup, Range_test1) {
...
     CHECK_EQUAL(5, v1[0]);
+}
+
+TEST(UtilsTestsGroup, intToPaddedString_test)
+{
+    CHECK_EQUAL("01", intToPaddedString(1,2));
+    CHECK_EQUAL("0100", intToPaddedString(100,4));
+    CHECK_EQUAL("100", intToPaddedString(100,2));
+    CHECK_EQUAL("1", intToPaddedString(1,1));
+    CHECK_EQUAL("0000056", intToPaddedString(56,7));
```
Comment (andrist) [inline test/utils_test.cpp:151]: can you add the corner case number=0 and a negative example to the tests?

**Review by andrist — APPROVED** [review]

**Review by Anatoliy-Litvinenko — APPROVED** [review]

File: test/utils_test.cpp (hunk 2 above)
Comment (adelebai) [reply test/utils_test.cpp:151]: negative example behaviour is undefined so probably shouldn't enforce a test on that. Can do 0 edge case.

File: util.hpp (hunk 1 above)
Comment (adelebai) [reply util.hpp]: thanks a lot! this is what happens when I write things at 11pm :O

File: test/utils_test.cpp (hunk 2 above)
Comment (andrist) [reply test/utils_test.cpp:151]: me (personally) I would add an assert(x>=0) if the behavior is undefined; I would prefer to know where/why things went wrong. Up to you; not sure if it fits the project style.
